sky/clouds/vast.py

```diff
@@ -0,0 +1,296 @@
+""" Vast Cloud. """
+
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple, Union
+
+from sky import clouds
+from sky.clouds import service_catalog
+from sky.utils import resources_utils
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib
+
+_CREDENTIAL_FILES = [
+    'config.toml',
+]
+
+
+@clouds.CLOUD_REGISTRY.register
+class Vast(clouds.Cloud):
+    """ Vast GPU Cloud
+
+    _REPR | The string representation for the Vast GPU cloud object.
+    """
+    _REPR = 'Vast'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE:
+            ('Spot is not supported, as vast API does not implement spot.'),
+        clouds.CloudImplementationFeatures.MULTI_NODE:
+            ('Multi-node not supported yet, as the interconnection among nodes '
+             'are non-trivial on Vast.'),
```

Comment:
Just want to make sure, is multi-node not supported in vast ai?

Reply:
Some of our customers have achieved it and we are currently in development for offering it as a formal service but we don't have it yet.